cmd/restic/cmd_version.go

```diff
@@ -21,8 +22,29 @@ Exit status is 0 if the command was successful, and non-zero if there was any er
 `,
 	DisableAutoGenTag: true,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Printf("restic %s compiled with %v on %v/%v\n",
-			version, runtime.Version(), runtime.GOOS, runtime.GOARCH)
+		if globalOptions.JSON {
+			type jsonVersion struct {
+				Version   string `json:"version"`
+				GoVersion string `json:"go_version"`
+				GoTarget  string `json:"go_target"`
```

Comment:
We discussed this a bit and would like to replace `go_target` with two separate keys instead, `go_os` and `go_arch` (which corresponds to `runtime.GOOS` and `runtime.GOARCH`). This way, users can more easily get to the individual pieces of data instead of having to split it up separately.